lib/lrama/state.rb

```diff
@@ -142,5 +160,274 @@ def rr_conflicts
         conflict.type == :reduce_reduce
       end
     end
+
+    def propagate_lookaheads(next_state)
+      next_state.kernels.map {|item|
+        lookahead_sets =
+          if item.position == 1
```

Comment:
[nits]: I personally like to make the condition order to be aligned with the original paper for readability of the algorithm. For example 

```ruby
if item.position > 1
  ...
else
  ...
end
```